docs/ARCHITECTURE.md

```diff
@@ -0,0 +1,77 @@
+
+In the future, the reverse proxy might integrate more tightly with the data proxying facilities in Habitat.
+
+#### Distributed File System
+Habitat will eventually come packaged with a distributed file system, resembling IPFS. The most important featured of this system is content addressing, which creates a number of benefits for running the filesystem in a distributed environment.
+
+Until the distributed file system can be implemented, an instance of IPFS is run alongside Habitat. 
+
+#### Process Manager
+The process manager allows for applications to be started and stopped on a Habitat node. When a process is started, a child process is created and added to the same process group as the Habitat server. If the Habitat server is killed, all processes will also be destroyed.
+
+In the future, we should consider whether this functionality should mostly be handled by systemd (and equivalent on other OSes).
+
+#### Data Sources
+To better facilitate data sharing across nodes, Habitat communities can publish "data sources" that conform to a specific schema specified in a schema registry. Depending on permissions, these data sources can be publicly available, allowing for the creation of aggregating applications, which can range from social networks to feed readers. 
+
+Since the ultimate source of data might not be contained by a node fielding a request, the `data proxy` can direct requests to the node that is able to serve teh correct data.
+
+### Control
+
+There are two systems that control Habitat nodes. The first is the `habitatctl` client, which directly accesses the `Habitat API` to perform actions. The second, and more important, is the community management service.
+
+#### Habitat Client
+The `habitatctl` CLI can be used to control Habitat remotely or locally. It provides commands that interface directly with the Habitat API. It gives the option for custom addresses, as well as the ability to hit the API via `libp2p` to circumvent NATs.
+
+In the future, a more interactive CLI can be created using the Bubble Tea library. In addition, `habitatctl` can run a local server for a client web app to make interacting with Habitat easier for non-technical users.
+
+#### Community Management
+A community is a cluster of nodes that collectively maintain a shared filesystem, shared applications, and more. These nodes can be spread out geographically and can number from one to thousands.
+
+##### State Management
+Community state is represented as a JSON object conforming to a JSON schema. There are a certain set of allowed state transitions that can be applied to the state object using JSON patches. Instances of the `Executor` interface receive these state updates via a channel, and can act on them. Each node service should implement this interface to be able to react to these updates.
```

Comment:
would be nice to give an example of state transitions or link to them in the code